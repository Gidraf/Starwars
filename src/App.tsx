import logo from './logo.svg';
import React, { Suspense } from 'react';
import Container from './components/Container';
import styled from 'styled-components';

const StyledApp = styled.div`
  text-align: center;
  overflow: hidden;
`;

const App = () => {
  return (
    <StyledApp>
      <Container />
    </StyledApp>
  );
};

export default App;
