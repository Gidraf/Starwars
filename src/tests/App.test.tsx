import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import App from '../App';
import { Provider } from 'react-redux';

describe('Snapshot Testing', () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store;

  it('It renders Successfully"', () => {
    store = mockStore(initialState);
    const { getByText } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const tree = renderer.create( <Provider store={store}>
      <App />
    </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
