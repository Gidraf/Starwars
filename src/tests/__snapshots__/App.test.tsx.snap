// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It Renders Successfully Shows "Hello world!" 1`] = `
<div
  className="sc-hAZoDl fyWgUn"
>
  <div
    className="sc-ftvSup FQGCe"
  >
    <img
      className="sc-kDDrLX bFihDq"
      src="starwars.jpeg"
    />
    <div
      className="sc-iqcoie gefhIs"
    >
      <h2
        className="sc-jqUVSM cxRjQe"
      >
        StarWars Films Search
      </h2>
      <input
        className="sc-papXJ kaRkdn"
        onChange={[Function]}
        placeholder="Search for a Film  🔍"
        value=""
      />
      <button
        className="sc-breuTD jjQvlr"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;

exports[`Snapshot Testing It renders Successfully" 1`] = `
<div
  className="sc-hAZoDl fyWgUn"
>
  <div
    className="sc-ftvSup FQGCe"
  >
    <img
      className="sc-kDDrLX bFihDq"
      src="starwars.jpeg"
    />
    <div
      className="sc-iqcoie gefhIs"
    >
      <h2
        className="sc-jqUVSM cxRjQe"
      >
        StarWars Films Search
      </h2>
      <input
        className="sc-papXJ kaRkdn"
        onChange={[Function]}
        placeholder="Search for a Film  🔍"
        value=""
      />
      <button
        className="sc-breuTD jjQvlr"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;

exports[`t Matches Snapshots Shows "Hello world!" 1`] = `
<div
  className="sc-hAZoDl fyWgUn"
>
  <div
    className="sc-ftvSup FQGCe"
  >
    <img
      className="sc-kDDrLX bFihDq"
      src="starwars.jpeg"
    />
    <div
      className="sc-iqcoie gefhIs"
    >
      <h2
        className="sc-jqUVSM cxRjQe"
      >
        StarWars Films Search
      </h2>
      <input
        className="sc-papXJ kaRkdn"
        onChange={[Function]}
        placeholder="Search for a Film  🔍"
        value=""
      />
      <button
        className="sc-breuTD jjQvlr"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
